File: practicas/biblio/biblioteca.py
from horario import Horario
from prestamo import Prestamo

class Biblioteca:
    def __init__(self, nombre, horario):
        self.nombre = nombre
        self.horario = horario  
        self.libros = []        
        self.autores = []       
        self.prestamos = []

    def agregar_libro(self, libro):
        self.libros.append(libro)

    def agregar_autor(self, autor):
        self.autores.append(autor)

    def prestar_libro(self, estudiante, libro):
        from prestamo import Prestamo
        prestamo = Prestamo(estudiante, libro)
        self.prestamos.append(prestamo)
        print(f"Libro prestado exitosamente: {libro.titulo}")

    def mostrar_estado(self):
        print(f"\n--- Biblioteca: {self.nombre} ---")
        self.horario.mostrar_horario()

        print("\nLibros disponibles:")
        for libro in self.libros:
            print(f"- {libro.titulo}")

        print("\nAutores registrados:")
        for autor in self.autores:
            autor.mostrar_info()

        print("\nPrestamos activos:")
        for prestamo in self.prestamos:
            prestamo.mostrar_info()
            print()

    def cerrar_biblioteca(self):
        self.prestamos.clear()
        print(f"La biblioteca {self.nombre} ha cerrado. Todos los prestamos han sido eliminados.")

File: practicas/bibli/biblioteca.java
package practicas.bibli;

import java.util.ArrayList;

public class biblioteca {
    private String nombre;
    private ArrayList<libro> libros;     
    private ArrayList<autor> autores; 
    private ArrayList<prestamo> prestamos;
    private horario horario;              

    public biblioteca(String nombre, horario horario) {
        this.nombre = nombre;
        this.horario = horario;
        this.libros = new ArrayList<>();
        this.autores = new ArrayList<>();
        this.prestamos = new ArrayList<>();
    }
    // Métodos para agregar libros
    public void agregarLibro(libro libro) {
        libros.add(libro);
    }
    // Métodos para agregar autores
    public void agregarAutor(autor autor) {
        autores.add(autor);
    }
    // Método para prestar libros
    public void prestarLibro(estudiante estudiante, libro libro) {
        prestamo p = new prestamo(estudiante, libro);
        prestamos.add(p);
        System.out.println("Libro prestado exitosamente: " + libro.getTitulo());
    }
    // Método para mostrar el estado de la biblioteca
    public void mostrarEstado() {
        System.out.println("\n--- Biblioteca: " + nombre + " ---");
        horario.mostrarHorario();

        System.out.println("\nLibros disponibles:");
        for (libro l : libros) {
            System.out.println("- " + l.getTitulo());
        }

        System.out.println("\nAutores registrados:");
        for (autor a : autores) {
            a.mostrarInfo();
        }

        System.out.println("\nPrestamos activos:");
        for (prestamo p : prestamos) {
            p.mostrarInfo();
            System.out.println();
        }
    }
    // Método para cerrar la biblioteca
    public void cerrarBiblioteca() {
        prestamos.clear(); 
        System.out.println("La biblioteca " + nombre + " ha cerrado. Todos los prestamos han sido eliminados.");
    }
}
